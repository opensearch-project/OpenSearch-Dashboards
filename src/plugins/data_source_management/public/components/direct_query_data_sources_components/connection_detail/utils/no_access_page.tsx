/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton, EuiEmptyPrompt, EuiPage, EuiText } from '@elastic/eui';
import React from 'react';

export const NoAccess = () => {
  return (
    <EuiPage>
      <EuiEmptyPrompt
        iconType="alert"
        title={<h2>{'No permissions to access'}</h2>}
        body={
          <EuiText>
            {
              'You are missing permissions to view connection details. Contact your administrator for permissions.'
            }
          </EuiText>
        }
        actions={
          <EuiButton color="primary" fill onClick={() => (window.location.hash = '')}>
            Return to data connections
          </EuiButton>
        }
      />
    </EuiPage>
  );
};
