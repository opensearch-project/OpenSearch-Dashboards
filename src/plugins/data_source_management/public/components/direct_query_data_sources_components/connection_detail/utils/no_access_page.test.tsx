/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow } from 'enzyme';
import { NoAccess } from './no_access_page';
import { EuiEmptyPrompt, EuiButton, EuiText } from '@elastic/eui';

describe('NoAccess', () => {
  const shallowComponent = () => shallow(<NoAccess />);

  test('renders correctly', () => {
    const wrapper = shallowComponent();
    expect(wrapper).toMatchSnapshot();
  });

  test('displays the correct icon type', () => {
    const wrapper = shallowComponent();
    expect(wrapper.find(EuiEmptyPrompt).prop('iconType')).toEqual('alert');
  });

  test('displays the correct title', () => {
    const wrapper = shallowComponent();
    expect(wrapper.find(EuiEmptyPrompt).prop('title')).toEqual(
      <h2>{'No permissions to access'}</h2>
    );
  });

  test('displays the correct body text', () => {
    const wrapper = shallowComponent();
    const bodyText = wrapper.find(EuiEmptyPrompt).prop('body') as React.ReactElement;
    const wrapperBody = shallow(<div>{bodyText}</div>);
    expect(wrapperBody.find(EuiText).dive().text()).toEqual(
      'You are missing permissions to view connection details. Contact your administrator for permissions.'
    );
  });

  test('renders a button with correct text and click behavior', () => {
    const wrapper = shallowComponent();
    const actions = wrapper.find(EuiEmptyPrompt).prop('actions') as React.ReactElement;
    const wrapperActions = shallow(<div>{actions}</div>);
    const button = wrapperActions.find(EuiButton);
    expect(button.prop('color')).toEqual('primary');
    expect(button.prop('fill')).toBe(true);
    expect(button.children().text()).toEqual('Return to data connections');

    // Simulate click
    const originalHash = window.location.hash;
    window.location.hash = '#test';
    button.simulate('click');
    expect(window.location.hash).toBe('');
    window.location.hash = originalHash; // Restore original hash
  });
});
