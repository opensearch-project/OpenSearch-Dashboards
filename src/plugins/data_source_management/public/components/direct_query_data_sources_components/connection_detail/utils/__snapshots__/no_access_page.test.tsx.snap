// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoAccess renders correctly 1`] = `
<EuiPage>
  <EuiEmptyPrompt
    actions={
      <EuiButton
        color="primary"
        fill={true}
        onClick={[Function]}
      >
        Return to data connections
      </EuiButton>
    }
    body={
      <EuiText>
        You are missing permissions to view connection details. Contact your administrator for permissions.
      </EuiText>
    }
    iconType="alert"
    title={
      <h2>
        No permissions to access
      </h2>
    }
  />
</EuiPage>
`;
